plugin/src/main/java/org/opensearch/ml/action/profile/MLProfileTransportAction.java

```diff
@@ -103,27 +112,45 @@ private MLProfileNodeResponse createMLProfileNodeResponse(MLProfileRequest mlPro
         MLProfileInput mlProfileInput = mlProfileRequest.getMlProfileInput();
         Set<String> targetModelIds = mlProfileInput.getModelIds();
+
+        boolean isSuperAdmin = isSuperAdminUserWrapper(clusterService, client);
+        Set<String> hiddenModels = mlProfileRequest.getHiddenModelIds() != null
+            ? mlProfileRequest.getHiddenModelIds()
+            : Collections.emptySet();
+        log.info("List of hidden models from Profile transport action: {}", Arrays.toString(hiddenModels.toArray()));
         Arrays.stream(mlTaskManager.getAllTaskIds()).forEach(taskId -> {
             MLTask mlTask = mlTaskManager.getMLTask(taskId);
-            if (mlProfileInput.isReturnAllTasks() || (!mlProfileInput.emptyTasks() && mlProfileInput.getTaskIds().contains(taskId))) {
-                log.debug("Runtime task profile is found for model {}", mlTask.getModelId());
-                mlLocalTasks.put(taskId, mlTask);
-            }
-            if (mlProfileInput.isReturnAllTasks() || (!mlProfileInput.emptyModels() && targetModelIds.contains(mlTask.getModelId()))) {
-                log.debug("Runtime task profile is found for model {}", mlTask.getModelId());
-                mlLocalTasks.put(taskId, mlTask);
+            if (isSuperAdmin || !hiddenModels.contains(mlTask.getModelId())) {
+                if (mlProfileInput.isReturnAllTasks() || (!mlProfileInput.emptyTasks() && mlProfileInput.getTaskIds().contains(taskId))) {
+                    log.debug("Runtime task profile is found for model {}", mlTask.getModelId());
+                    mlLocalTasks.put(taskId, mlTask);
+                }
+                if (mlProfileInput.isReturnAllTasks() || (!mlProfileInput.emptyModels() && targetModelIds.contains(mlTask.getModelId()))) {
+                    log.debug("Runtime task profile is found for model {}", mlTask.getModelId());
+                    mlLocalTasks.put(taskId, mlTask);
+                }
             }
         });
         Arrays.stream(mlModelManager.getAllModelIds()).forEach(modelId -> {
-            if (mlProfileInput.isReturnAllModels() || (!mlProfileInput.emptyModels() && targetModelIds.contains(modelId))) {
-                log.debug("Runtime model profile is found for model {}", modelId);
-                MLModelProfile modelProfile = mlModelManager.getModelProfile(modelId);
-                if (modelProfile != null) {
-                    mlLocalModels.put(modelId, modelProfile);
+            if (isSuperAdmin || !hiddenModels.contains(modelId)) {
+                if (mlProfileInput.isReturnAllModels() || (!mlProfileInput.emptyModels() && targetModelIds.contains(modelId))) {
+                    log.debug("Runtime model profile is found for model {}", modelId);
+                    MLModelProfile modelProfile = mlModelManager.getModelProfile(modelId);
+                    if (modelProfile != null) {
+                        if (isSuperAdmin && hiddenModels.contains(modelId)) {
```

Comment:
It's becoming a 4 layer nested If...else... At least the last two ifs can be merged into one.

Reply:
Yeah agree, let me simplify this.